naga/src/back/msl/writer.rs

```diff
@@ -3720,6 +3730,734 @@ impl<W: Write> Writer<W> {
+                                "{}return uint2((b0 << 24 | b1 << 16 | b2 << 8 | b3), \
+                                                (b4 << 24 | b5 << 16 | b6 << 8 | b7));",
+                                back::INDENT
+                            )?;
+                            writeln!(self.out, "}}")?;
+                            (name, 8)
+                        }
+                        Uint32x3 => {
+                            let name = self.namer.call("unpackUint32x3");
+                            writeln!(
+                                self.out,
+                                "uint3 {name}(uint b0, \
+                                              uint b1, \
+                                              uint b2, \
+                                              uint b3, \
+                                              uint b4, \
+                                              uint b5, \
+                                              uint b6, \
+                                              uint b7, \
+                                              uint b8, \
+                                              uint b9, \
+                                              uint b10, \
+                                              uint b11) {{"
+                            )?;
+                            writeln!(
+                                self.out,
+                                "{}return uint3((b0 << 24 | b1 << 16 | b2 << 8 | b3), \
+                                                (b4 << 24 | b5 << 16 | b6 << 8 | b7), \
+                                                (b8 << 24 | b9 << 16 | b10 << 8 | b11));",
+                                back::INDENT
+                            )?;
+                            writeln!(self.out, "}}")?;
+                            (name, 12)
+                        }
+                        Uint32x4 => {
+                            let name = self.namer.call("unpackUint32x4");
+                            writeln!(
+                                self.out,
+                                "uint4 {name}(uint b0, \
+                                              uint b1, \
+                                              uint b2, \
+                                              uint b3, \
+                                              uint b4, \
+                                              uint b5, \
+                                              uint b6, \
+                                              uint b7, \
+                                              uint b8, \
+                                              uint b9, \
+                                              uint b10, \
+                                              uint b11, \
+                                              uint b12, \
+                                              uint b13, \
+                                              uint b14, \
+                                              uint b15) {{"
```

Comment:
I was thinking it would be nicer to pass these as an array but I don't know how callers can get a sub-array from the initial array in MSL.

Reply:
I don't see how to do this.